src/transport/AuthenticatedSessionEstablishmentDelegate.h

```diff
@@ -0,0 +1,69 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the Delegate class that contains callbacks to
+ *      establish secure session pairing and update status of the pairing process.
+ *
+ */
+
+#pragma once
+
+#include <system/SystemPacketBuffer.h>
+#include <transport/raw/MessageHeader.h>
+#include <transport/raw/PeerAddress.h>
+
+namespace chip {
+
+class DLL_EXPORT AuthenticatedSessionEstablishmentDelegate
+{
+public:
+    /**
+     * @brief
+     *   Called when pairing session generates a new message that should be sent to peer.
+     *
+     * @param header the message header for the sent message
+     * @param peerAddress the destination of the message
+     * @param msgBuf the raw data for the message being sent
+     * @return CHIP_ERROR Error thrown when sending the message
+     */
+    virtual CHIP_ERROR SendPairingMessage(const PacketHeader & header, const Transport::PeerAddress & peerAddress,
```

Comment:
This needs a better name, but per discussion followup is OK, since that involves touching various existing code.  Please make sure there's an issue tracking the renaming and a TODO comment here pointing to that issue.